paddle/fluid/framework/attribute.h

```diff
@@ -208,22 +208,37 @@ Attribute GetAttrValue(const proto::OpDesc::Attr& attr_desc);
 
 class AttrReader {
  public:
-  explicit AttrReader(const AttributeMap& attrs) : attrs_(attrs) {}
+  explicit AttrReader(const AttributeMap& attrs)
+      : attrs_(attrs), attrs_default_(nullptr) {}
+
+  AttrReader(const AttributeMap& attrs, const AttributeMap& attrs_default)
+      : attrs_(attrs), attrs_default_(&attrs_default) {}
 
   template <typename T>
   inline const T& Get(const std::string& name) const {
-    PADDLE_ENFORCE_NE(attrs_.count(name), 0,
-                      platform::errors::NotFound(
-                          "Attribute (%s) should be in AttributeMap.", name));
+    auto it = attrs_.find(name);
+    if (it == attrs_.end()) {
+      if (attrs_default_ != nullptr) {
+        it = attrs_default_->find(name);
+        if (it == attrs_default_->end()) {
```

Comment:
bool found = false；

if ( it == attrs_ends() )
{
  if (attrs_default_ != nullptr) {
        it = attrs_default_->find(name);
        if (it == attrs_default_->end()) 
        {
            found = false;
         }
}